src/controller/expenses.controller.js

```diff
@@ -0,0 +1,122 @@
+const expensesService = require('../services/expenses.service');
+const userService = require('../services/user.service');
+
+const get = async (req, res) => {
+  const { userId, from, to, categories } = req.query;
+  const expenses = await expensesService.getAllExpenses();
+
+  if (!userId && !categories && (!from || !to)) {
+    return res.send(expenses);
+  }
+
+  let filteredExpenses = expenses;
+
+  if (userId) {
+    filteredExpenses = expensesService.filterExpensesById(
+      filteredExpenses,
+      userId,
+    );
+  }
+
+  if (categories) {
+    const normalizedCategories = expensesService.normalizeCategory(categories);
+
+    filteredExpenses = expensesService.filterExpensesByCategory(
+      filteredExpenses,
+      normalizedCategories,
+    );
+  }
+
+  if (from && to) {
+    const fromDate = new Date(from);
+    const toDate = new Date(to);
+
+    filteredExpenses = expensesService.filterExpensesByDate(
+      filteredExpenses,
+      fromDate,
+      toDate,
+    );
+  }
+
+  res.send(filteredExpenses);
+};
+
+const getOne = async (req, res) => {
+  const { id } = req.params;
+
+  const expense = await expensesService.getById(id);
+
+  if (!expense) {
+    res.sendStatus(404);
+
+    return;
+  }
+
+  res.send(expense);
+};
+
+const create = async (req, res) => {
+  const { userId, spentAt, title, amount, category, note } = req.body;
+
+  if (!userId || !spentAt || !title || !amount) {
```

Comment:
You are checking for the presence of certain fields in the request body but not checking their types or values. This could lead to unexpected errors or incorrect data being stored in the database. It would be a good idea to add some validation to ensure that the fields are of the expected types and meet any other constraints (e.g., 'amount' should be a positive number).